### ISSUE TITLE
Record "Demo Video" for social media

### ISSUE THREAD

**Issue by bhoov** [bhoov/manim-notebook#37]
Indeed, this extension just keeps getting cooler and cooler, in huge part thanks to @Splines awesome contributions over the past few days. I'm hoping this will gain a decent amount of traction online. We can help this along by making a "cool" demo video to accompany the announcement.

I do not consider myself good at production or advertising, but my lab at Georgia Tech has historically created some really nice videos to accompany some of the online tools we make. See the [Transformer Explainer Demo](https://www.youtube.com/watch?v=ECR4oAwocjs) or [GAM Coach](https://www.youtube.com/watch?v=ubacP34H9XE) or [Diffusion Explainer](https://www.youtube.com/watch?v=Zg4gxdIWDds) as examples.

Once all main features are finalized and merged, I'm happy to record a 45s-1min demo video for this tool to share on social media (time permitting in the next 4 days🤞 😭😭😭). Please help me refine which features to showcase/the order to present them/how best to describe them:

**Features to showcase** (note: I think it is appropriate to emphasize *interactivity* in this video, so we can include some of Grant's baked-in animations on the video previewer)
- Start developing from inside a python file (`Cmd+' Cmd+s`)
- With "Manim cells", render only the parts of a scene that you are developing (`Cmd+' Cmd+e`)
- Cells collapse and expand, just like code regions! (toggle arrow next to cell)
- Test a single line (`Cmd+' Cmd+r`)
- Test a selection of lines (Highlight and then `Cmd+' Cmd+r`)
- TODO: **Export current scene as mp4** (in writing up these features, I see this as a feature we can possibly add that will make Manim Notebook a full pipeline for making manim videos)
- Clear all objects from scene at any time (`Cmd+' Cmd+c`)
- Exit the scene (`Cmd+' Cmd+w`)
(Animation window)
- Pan and zoom around your scene
- Copy the current camera view
- Recenter the camera
- Select objects

Finally, what scene would be fun+catchy to demo all these awesome features? Should we just choose an example scene from the notebook? I am not sure we can re-use one of Grant's videos to showcase the tool, since our tool is MIT licensed and his videos are creative commons.

**Comment by Splines** [comment]
> I'm happy to record a 45s-1min demo video for this tool to share on social media

That'd be really awesome 🙌

> Start developing from inside a python file (Cmd+' Cmd+s)

While this command is useful, I wouldn't put this at the start of the video and instead focus on the preview cell feature first. This showcases how easy it is (just one button click).


### Some further storyboard ideas
- Show in a timelapse how Manim code is typed.
- Place cursor at top of code block. [Zoom in] Type `##` [Zoom out a bit] See how Manim Cell appears (e.g. border is now shown)
- Toggle folding ranges on music beat. [Fold, unfold, maybe even twice]
- [Zoom in] Click on `Preview Manim Cell`.
- [Cut or whip, close-up] Notification `Starting Manim...`. Beat builds.
- [Wide] ManimGL window opens. Beat continues to build.
- [Wide] ManimGL window is loaded. Music drops. Manim preview with nice animation is shown. We also see how progress notification is showing different parts of the scene.
- [Fluent text fade in, char by char] "Preview your Manim code. Easily.". Next line: "Manim Notebook for VSCode. Install now.".

### Showcased code
For copyright reasons and also for curtsy, I wouldn't use any code by 3b1b in the ad, not even sample scenes. It shows our originality if we come up with our own scene to showcase in an ad. E.g. something of the scope of the LorenzAttractor that Grant Sanderson showed, or smaller.

Then, we could even publish a video where we are going through how to build this exact scene. And while doing that use our extension along the way, i.e. incorporate it naturally in the flow of what we do. This offers the possibility to showcase many more commands that might not have fit into a short ad. And users will engage more with it because it's more a tutorial-like format where they might want to follow along anyways. This is not the case for an ad they randomly see on the Internet.

### Other IT-related ads
- For inspiration, I like the [Windows Terminal advertisement](https://apps.microsoft.com/detail/9n0dx20hk701) although of course we don't have the time/budget to build an ad comparable to that one. But the aesthetics are very appealing in my opinion.

**Comment by Splines** [comment]
What do you think about it? During the next weeks I will work on a 1-minute teaser video based on the ideas discussed in my previous comment. The showcased code in that video will be about drawing the evolute of a [nephroid](https://en.wikipedia.org/wiki/Nephroid). I want to publish that video mid-december (or earlier).

**Comment by Splines** [comment]
For the song, I think I will use [this one](https://www.epidemicsound.com/track/Mhe3rIPhaa/) and cut together a 40s version of it. I love the funky vibes in it.

**Comment by bhoov** [comment]
Super neat stuff @Splines. I'm not familiar with collaboratively storyboarding videos -- what kind of feedback is most helpful for you? Is [the video](https://youtu.be/Wp1EdddF6EI) representative of the final design (sans the comments)?

Re: the music track. I think it's important to be able to upload the video on all platforms. I'm happy to chip in the $40 to pay for the license. Shoot me an email and we can get that sorted out.

**Comment by Splines** [comment]
> Is the video representative of the final design (sans the comments)?

Hey, no not at all. So regarding the Manim output, yes, the curve is what I want to show in the end. But regarding the video itself: no not at all. I will animate all of this in a 3D world in Blender and put in recorded screen overlays as planes that get some reflective material to better integrate them in the world. I already indicate some camera movements in the storyboard though (and describe what I mean by 3D flying to the progress strip in the YT comments). The scene background will probably be black and I'm not sure yet whether I will show VSCode in light or dark theme...

> I'm happy to chip in the $40 to pay for the license.

Nice, thanks a lot for the offer. I could also contribute like $20. I will contact you about this in the final phase of the project.

**Comment by Splines** [comment]
That said, I don't have much experience with videos like this neither and just want to give it a try ;) The end product won't be comparable to an Apple ad of course, but will hopefully be more interactive and visually pleasing mere screen recordings with text overlays.

**Comment by Splines** [comment]
Some further inspirations. Me being an animation noob, I won't even get close to this, but at least one should think big, right ;)

- https://youtu.be/HrlNTIbo9o4
- https://youtu.be/Mei3QcQ66Mw -> this is just insane
- https://youtu.be/QFCV6EkqRQs?t=50 -> this is probably very close to what I think of doing sometimes in the video: having the VSCode editor placed in the 3D world, we can do these subtle camera movements where we move around the screen.

**Comment by bhoov** [comment]
> The end product won't be comparable to an Apple ad of course, but will hopefully be more interactive and visually pleasing mere screen recordings with text overlays.

Oh the mere ambition will make this an amazing video :). Paired with the fun math animations you already have in the video (maybe we can have a "text/LaTex" manipulation example as well?), I think it can look really good.

Some comments about the reference videos, since this helps me get a feel for what your vision is:

- https://youtu.be/HrlNTIbo9o4 :: The 3D effects don't add to (in fact, they possibly subtract from) the message here. It is hard for me to visualize using this as a "tool" on my 2D screen. My eyes don't have a cursor to follow so I never know where to look. They often enlarge pieces of the display and move them around so I don't have a feel for the tool overall (e.g., the [New Status Widget](https://youtu.be/HrlNTIbo9o4?si=pUKOTB_lc9WFhsi1&t=9) demo), and the interpolation between "scenes" is so fast to be jarring. 
- https://youtu.be/Mei3QcQ66Mw :: This one is a lot more polished, but the scenes definitely lean "flash" over "function". Since we use manim, which is a beautiful combination of form and function, we may not need to put this much effort in to make our "product" stick out. This video did a lot to try to make the boring-old email interface pop. We want to sell our "interaction features" more than our UI design.
- https://youtu.be/Mei3QcQ66Mw :: Despite the dated UI, this video is actually really good at communicating what the product can do. It would be better if the sound clip was altered live as they were changing the settings so you could see the large effect simple button clicks could have. We actually can do that very easy since our output is an eye catching video animation of good-looking math
- https://youtube.com/shorts/77utm55_7EE :: !! That's CRAZY. Definitely eye catching 😂

**Comment by VladimirFokow** [comment]
ahm.. not to spoil the fun, but I was imagining
- no-sound (maybe a gif, or several gifs - each focused on a single feature)
- the fewest diffs with the 3b1b's example in his video
- no moving of the screen. Just clean and maximally simple showcase of the workflow for the new users of this extension

**Comment by Splines** [comment]
> Just clean and maximally simple showcase of the workflow

Yeah sure, the Readme is another format and maybe even just static images would be better there. But for me that doesn't rule out having a separate advertising video. The ad is not intended to showcase every single feature in a rigorous way, **instead see it as an appetizer for people stumbling upon it through social media**. When they get curious enough, they will check out the extension on their own and get to know the features anyways (e.g. through good Readme documentation and through an extension walkthrough #40 and the command palette).

And even though the Readme will serve as place to document all features, it could also host, at the top, our ad video, just for a first fun demonstration.

And then, I'd like to do the video to also level up my animation skills a bit and to get creative in other ways that pure coding allows me ;)

> Despite the dated UI

I think you mixed up a link there and meant Ableton Live? Haha, I was chuckling when reading this since it's the UI of their flagship product ;)

> (maybe we can have a "text/LaTex" manipulation example as well?),

Sure, maybe we can add a slider that shows how a parameter of the curve is changing dynamically. And add text to it indicating the value of that param, e.g. `a = 3.14`).

> We want to sell our "interaction features" more than our UI design.

Yeah, that's true, I try to find a good balance with a bit "shiny" in the beginning, than settling down to show more of how everything works together without cutting/zooming too much.

> My eyes don't have a cursor to follow so I never know where to look.

My idea was to add a 3D cursor right from the beginning, maybe in the style of [this one](https://youtu.be/_rb31gBRaY8). It could greatly help guide the viewer's focus.

**Comment by bhoov** [comment]
@VladimirFokow I believe the video @Splines is working on is for the "social media" push to get people interested in this extension (please correct me if I'm wrong). I expect the final product of the video to look nice and not be too overwhelming (e.g., be more in the style of video in bullet 3 from  [this comment](https://github.com/Manim-Notebook/manim-notebook/issues/37#issuecomment-2468480726) ). Do you see any issues with including this social media video in the readme? If you would like, it's very easy to include the more "technical" gif in the manner you describe

**Comment by Splines** [comment]
> for the "social media" push to get people interested in this extension

Yes, exactly 👍 This is what I meant with: _"instead see it as an appetizer for people stumbling upon it through social media."_

---

By the way, I'm playing around a bit more with parametric equations and found this one to produce the illusion of a donut. But unfortunately, the evolute is not too nice anymore...


https://github.com/user-attachments/assets/06cd150b-73de-4c84-9b65-7f8775ed9ccd

**Comment by Splines** [comment]
Or maybe something like this:

https://github.com/user-attachments/assets/7d06d0c3-6de1-4575-a190-35885d8688ba
